djpsa/halo/sync.py

```diff
@@ -41,13 +41,42 @@ def _unpack_records(self, response):
         return records
 
 
+class CreateMixin:
+
+    def create(self, data, *args, **kwargs):
+        body = self._convert_fields_to_api_format(data)
+        return self.client.create(body)
+
+
+class UpdateMixin:
+
+    def update(self, record_id, data, *args, **kwargs):
```

Comment:
As discussed- make this update the model using update_or_create. Probably it should take the model as an argument, not an integer ID.

Probably it should not save the model itself, but let the caller save it?

Reply:
No, it shouldn't give the caller a chance to not save it. You would always want it saved as its already been updated in the remote PSA. Not saving would just de-sync the record.